x/bank/simulation/msgs.go

```diff
@@ -13,47 +13,33 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/auth"
 	"github.com/cosmos/cosmos-sdk/x/bank"
 	"github.com/cosmos/cosmos-sdk/x/mock"
-	"github.com/cosmos/cosmos-sdk/x/mock/simulation"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
 )
 
 // SendTx tests and runs a single msg send where both
 // accounts already exist.
-func SendMsg(mapper auth.AccountKeeper, bk bank.Keeper) simulation.Operation {
+func SimulateMsgSend(mapper auth.AccountKeeper, bk bank.Keeper) simulation.Operation {
 	handler := bank.NewHandler(bk)
-	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account, event func(string)) (action string, fOps []simulation.FutureOperation, err error) {
-		fromAcc, action, msg, abort := createSendMsg(r, ctx, accs, mapper)
-		if abort {
-			return action, nil, nil
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account, event func(string)) (
+		opMsg simulation.OperationMsg, fOps []simulation.FutureOperation, err error) {
+
+		fromAcc, comment, msg, ok := createMsgSend(r, ctx, accs, mapper)
+		opMsg = simulation.NewOperationMsg(msg, ok, comment)
+		if !ok {
+			return opMsg, nil, nil
 		}
 		err = sendAndVerifyMsgSend(app, mapper, msg, ctx, []crypto.PrivKey{fromAcc.PrivKey}, handler)
 		if err != nil {
-			return "", nil, err
+			return opMsg, nil, err
```

Comment:
Can we log an event regardless ("bank/sendAndVerifyTxSend/failure" or the like, something standardized)?

Reply:
yeah I just added a unified approach (see the commit) let me know what you think

Reply:
Looks good :+1: